iOS/DuckDuckGo/Fire/FireConfirmationViewModel.swift

```diff
@@ -35,18 +37,62 @@ class FireConfirmationViewModel: ObservableObject {
     
     let onConfirm: () -> Void
     let onCancel: () -> Void
-    let showAIChatsOption: Bool = true
     
     // MARK: - Private Variables
-    private let tabsModel: TabsModeling?
-    private let historyManager: HistoryManaging?
+    private let tabsModel: TabsModeling
+    private let historyManager: HistoryManaging
     private let tld: TLD
-    private let fireproofing: Fireproofing?
+    private let fireproofing: Fireproofing
+    private let aiChatSettings: AIChatSettingsProvider
     
-    init(tabsModel: TabsModeling?,
-         historyManager: HistoryManaging?,
+    // MARK: - Lazy Variables
+
+    private lazy var sitesCount: Int = {
+        return self.computeNonFireproofedDomainCount()
+    }()
+    
+    private lazy var tabsCount: Int = {
+        guard tabsModel.hasActiveTabs else {
+            return 0
+        }
+        return tabsModel.count
+    }()
+    
+    // MARK: - Computed Properties
+    
+    var isDeleteButtonDisabled: Bool {
+        !clearTabs && !clearData && !clearAIChats
+    }
+    
+    var isClearTabsDisabled: Bool {
+        tabsCount == 0
+    }
+    
+    var isClearDataDisabled: Bool {
+        return historyManager.isEnabledByUser && sitesCount == 0
+    }
+    
+    var showAIChatsOption: Bool {
+        aiChatSettings.isAIChatEnabled
+    }
+    
+    // MARK: - Persistence Storage
+    @UserDefaultsWrapper(key: .fireConfirmationClearTabs, defaultValue: true)
+    private var storedClearTabs: Bool
+    
+    @UserDefaultsWrapper(key: .fireConfirmationClearData, defaultValue: true)
+    private var storedClearData: Bool
+    
+    @UserDefaultsWrapper(key: .fireConfirmationClearAIChats, defaultValue: false)
+    private var storedClearAIChats: Bool
```

Comment:
We're trying to move away from UserDefaults and this wrapper, see https://app.asana.com/1/137249556945/project/1202500774821704/task/1210379859420469?focus=true